### PR TITLE
Login

### PR DESCRIPTION
**Pull request by Yeganeh434**

### CODE REVIEW

File: handler.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	hasher.Write([]byte(input))
+	hashedBytes := hasher.Sum(nil)
+	hashedString := hex.EncodeToString(hashedBytes)
+	return hashedString
+}
+
+func authenticateUser(c *gin.Context) {
+	var loginData loginBody
+	err := c.BindJSON(&loginData)
+
```
Comment (mhghw) [inline handler.go]: unnecessary new-line

File: handler.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	if err != nil {
+		fmt.Errorf("error binding JSON:%w", err)
+		c.Status(400)
+		return
+	}
+
+	if len(loginData.Username) < 3 || len(loginData.Password) < 8 {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"error": "the username or password is incorrect",
```
Comment (mhghw) [inline handler.go]: Use custom errors.
You can do that by defining `var ErrIncorrectUsernameOrPassword=errors.New("the username or password is incorrect")`.
It's not necessary but it's more clear for the reader.

Also you can create an object for your errors, define it like
```
type HTTPError struct{
Error error `json:"error"`
}
```
you can marshal it using a json.Marshal function to a json

File: handler.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+		})
+		return
+	}
+
+	//checking entered data with data that is already stored
+	userUnderReveiw, err1 := db.UsersDB.GetUserByUsername(loginData.Username)
```
Comment (mhghw) [inline handler.go]: what does err1 means?
and why don't use err instead of err1?

File: login.go [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package api
+
+import (
+	"github.com/gin-gonic/gin"
+)
+
+func StartWebServer() {
```
Comment (mhghw) [inline login.go:16]: Where is port?

**Review by mhghw — CHANGES_REQUESTED** [review]

File: handler.go (hunk 3 above)
Comment (Yeganeh434) [reply handler.go]: I used 'err1' because I had defined 'err' above. is it okay to define a variable twice?

File: login.go (hunk 4 above)
Comment (Yeganeh434) [reply login.go:16]: you mean the port that we put in front of this  'router.Run()'  funtion?
should I use port 8080 or another port?

File: handler.go (hunk 3 above)
Comment (mhghw) [reply handler.go]: For err, yes.

File: login.go (hunk 4 above)
Comment (mhghw) [reply login.go:16]: You need to read a port from terminal and then run the server on that port, if no port was provided, use a default port

**Review by mhghw — APPROVED** [review]

**Pull request merged**